src/components/buttonsIndicators/button/index.tsx

```diff
@@ -14,19 +15,16 @@ import {
 export interface Props {
   text: string
   size?: Size
-  type?: Type
   brand?: Brand
   level?: Level
-  main?: boolean
   onClick?: () => void
   id?: string
   disabled?: boolean
   icon?: {image: React.ReactNode, position: 'before' | 'after'}
   className?: string
 }
 
-export type Level = 'primary' | 'secondary' | 'tertiary'
-export type Type = 'default' | 'accent' | 'warn' | 'subtle'
+export type Level = 'default' | 'primary' | 'secondary' | 'tertiary'
```

Comment:
I'm not sure it's clear to me that we should have a `default` value here - it's not understandable how this semantically fits in between the hierarchy of primary / secondary / tertiary

Reply:
I just took that directly from abstract, cc @karenkliu

Reply:
@karenkliu @antonok-edm we should probably sync-up as it seems that whilst this has removed some variations, it seems more complicated to use. Right now we have properties that easily allow a developer or designer to know when to use which button:

- 'level' which indicates the importance of the button (most important in the UI vs one of many equal actions) **and affects the style of the button**,

![image](https://user-images.githubusercontent.com/741836/65189405-ad70b280-da25-11e9-9228-2f489e148890.png)


- 'type' which indicates the type of action (positive, negative, primary brand action) **and just really affects the color**

![image](https://user-images.githubusercontent.com/741836/65190676-2fada680-da27-11e9-8555-02d32d96f599.png)

These two properties are independent and could allow for all combinations.

In this new version we have a single property called 'Levels' in which I would expect the values to be hierarchical and has 4 variations - `primary`, `secondary`, `tertiary` and `default`. I wouldn't know when to use 'default' vs the hierarchical values. I would expect most buttons which aren't Primary actions to be Secondary actions, but now I'm guessing they should be Default actions and it's unclear when to use Secondary vs Tertiary.

![image](https://user-images.githubusercontent.com/741836/65191001-7a2f2300-da27-11e9-8b21-4023d15c7b33.png)

IMO we shouldn't mix metaphors here - either keep the semantic definitions (and ideally backwards-compatible) or choose new ones.


As another aside, these new button styles are slightly different to the other new button styles that Ross implemented in brave-core's WebUI at https://github.com/brave/brave-core/pull/2778. Those ones have a slight hover effect on 'pressed' and also a nicer focus outline style (that last one we should definitely adopt).